xds/src/test/java/io/grpc/xds/ServerXdsClientNewServerApiTest.java

```diff
@@ -723,6 +723,28 @@ static Listener buildListenerWithDefaultFilterChain(String name, int portValue,
             .build();
   }
 
+  static Listener buildListenerWithFilterChain(String name, int portValue, String address,
```

Comment:
As we are merging into a single XdsClient implementation, this class should be eliminated as well. Try move code close to where it is used.

Reply:
Yes, this will happen once I have eliminated `ServerXdsClient` which is not part of this PR. Once `ServerXdsClient` *and* v2 code path is eliminated a lot of these things will go away

Reply:
No, you should do it in this PR. Writing (good) test is part of the implementation. You should not just put code in random places and may (or may not) come back to do clean up separately. There is no reason to first add some bad code and then do it later a second time to rewrite it.

Reply:
My next PR is going to get rid of all ServerXds* classes so I will have to move the required code closer to where it is used. There is no point in front loading all the work in this PR which defeats the purpose of splitting work in multiple PRs.